src/main/kotlin/net/atos/zac/app/zaak/ZaakRestService.kt

```diff
@@ -279,23 +281,22 @@ class ZaakRestService @Inject constructor(
                 AANMAKEN_ZAAK_REDEN
             )
         }
-        // if BPMN support is enabled and if a referentieproces is defined for the zaaktype, start a BPMN process
-        // note that we misuse the referentieproces-name field to indicate that we use BPMN for a certain zaaktype
-        // however this needs to be changed because this field is actually filled for all our 'CMMN' zaaktypes currently..
-        if (configuratieService.featureFlagBpmnSupport() && zaaktype.referentieproces?.naam?.isNotEmpty() == true) {
+        // if BPMN support is enabled and a BPMN process definition is defined for the zaaktype, start a BPMN process;
+        // otherwise start a CMMN case
+        val processDefinition = bpmnService.findProcessDefinitionForZaaktype(zaaktypeUUID)
```

Comment:
I think it would be nice(er) to determine the type of service that will handle the case instead of only checking if it is a BPMN process. Then the next step is invoking that service and let it retrieve what it needs to handle it...

Reply:
I'm not sure what you mean exactly? You mean that the `ZaakRestService` itself should determine whether the zaaktype is a CMMN or a BPMN zaaktype? That would mean that the `ZaakRestService` would need to have all the logic to do so and that is quite a lot (including performing queries on our database). That's not really feasible. 

I think logically you would want to add another layer of abstraction in between the `ZaakRestService` and the CMMN and BPMN services that could handle this kind of stuff. That's way too much for this PR though and I see that as part of the BPMN epics in which we need to revisit all this code anyhow.

Reply:
So I propose to leave this for now (this PR) and that we refactor and move this logic in a later story. Ok @bas-info-nl ?

Reply:
Yes, fine to leave this.